python/lsst/ts/observatory/control/base_calsys.py

```diff
@@ -159,6 +159,32 @@ async def run_calibration_sequence(
         """
         raise NotImplementedError()
 
+    @abc.abstractmethod
```

Comment:
If this is going to be an abstract method, you will also need to add it to `ATCalsys`. You can leave it not implemented for now, just do:

```
    async def calculate_optimized_exposure_times(
        self, sequence_name: str
    ) -> dict[str, float]:
    """ <Docstring>
    """
    raise NotImplementedError()
```